[Airflow] update containerized setup instructions (#9707)
DataDog/integrations-core: airflow/README.md

#### Containerized

##### Configure Datadog Agent Airflow Integration

For containerized environments, see the [Autodiscovery Integration Templates][13] for guidance on applying the parameters below.

| Parameter            | Value                 |
|----------------------|-----------------------|
| `<INTEGRATION_NAME>` | `airflow`             |
| `<INIT_CONFIG>`      | blank or `{}`         |
| `<INSTANCE_CONFIG>`  | `{"url": "http://%%host%%"}` |

##### Connect Airflow to DogStatsD

Connect Airflow to DogStatsD (included in the Datadog Agent) by using the Airflow `statsd` feature to collect metrics. For more information about the metrics reported by the Airflow version used and the additional configuration options, see the Airflow documentation below:
- [Airflow Metrics Documentation][16]
- [Airflow Metrics Configuration Documentation][17]

**Note**: Presence or absence of StatsD metrics reported by Airflow might vary depending on the Airflow Executor used. For example: `airflow.ti_failures/successes, airflow.operator_failures/successes, airflow.dag.task.duration` are [not reported for `KubernetesExecutor`][13]. 

> **Note**: The environment variables used for Airflow may differ between versions, please consult the appropriate version relative to your Airflow deployments. For example in Airflow `2.0.0` this utilizes the environment variable `AIRFLOW__METRICS__STATSD_HOST`, whereas Airflow `1.10.15` utilizes `AIRFLOW__SCHEDULER__STATSD_HOST`. 

Reviewer: ```suggestion
**Note**: The environment variables used for Airflow may differ between versions. For example in Airflow `2.0.0` this utilizes the environment variable `AIRFLOW__METRICS__STATSD_HOST`, whereas Airflow `1.10.15` utilizes `AIRFLOW__SCHEDULER__STATSD_HOST`. 
```

The Airflow StatsD configuration can be enabled with the following environment variables in a Kubernetes Deployment:
  ```yaml
  env:
    - name: AIRFLOW__SCHEDULER__STATSD_ON
      value: "True"
    - name: AIRFLOW__SCHEDULER__STATSD_PORT
      value: "8125"
    - name: AIRFLOW__SCHEDULER__STATSD_PREFIX
      value: "airflow"
    - name: AIRFLOW__SCHEDULER__STATSD_HOST
      valueFrom:
        fieldRef:
          fieldPath: status.hostIP
  ```

> Where the environment variable for the host endpoint `AIRFLOW__SCHEDULER__STATSD_HOST` is supplied with the Host IP address of the node to route the StatsD data to the Datadog Agent pod on the same node as the Airflow pod. This setup also requires the Agent to have a `hostPort` open for this port `8125` and accepting non-local StatsD traffic. For more information on this see [DogStatsD on Kubernetes Setup here][19].

Reviewer: ```suggestion
The environment variable for the host endpoint `AIRFLOW__SCHEDULER__STATSD_HOST` is supplied with the node's host IP address to route the StatsD data to the Datadog Agent pod on the same node as the Airflow pod. This setup also requires the Agent to have a `hostPort` open for this port `8125` and accepting non-local StatsD traffic. For more information, see [DogStatsD on Kubernetes Setup here][19].
```
